core/src/utils/animation/animation.ts

```diff
@@ -0,0 +1,816 @@
+      progressStep(1 - step);
+    }
+
+    if (!parentAnimation) {
+      play();
+    }
+
+    return generatePublicAPI();
+  };
+
+  const pause = (): Animation => {
+    childAnimations.forEach(animation => {
+      animation.pause();
+    });
+
+    if (initialized) {
+      if (supportsWebAnimations()) {
+        webAnimations.forEach(animation => {
+          animation.pause();
+        });
+      } else {
+        elements.forEach(element => {
+          (element as HTMLElement).style.animationPlayState = 'paused';
+        });
+      }
+    }
+
+    return generatePublicAPI();
+  };
+
+  const playAsync = (): Promise<Animation> => {
+    return new Promise(resolve => {
+      onFinish(resolve);
+      play();
+
+      return generatePublicAPI();
+    });
+  };
+
+  const playSync = (): Animation => {
+    shouldForceSyncPlayback = true;
+
+    onFinish(() => shouldForceSyncPlayback = false);
+    play();
+
+    return generatePublicAPI();
+  };
+
+  const play = (): Animation => {
+    childAnimations.forEach(animation => {
+      animation.play();
+    });
+
+    if (!initialized) {
+      initializeAnimation();
+    }
+
+    if (supportsWebAnimations()) {
+      webAnimations.forEach(animation => {
+        animation.play();
+      });
+    } else {
+      elements.forEach(element => {
+        if (_keyframes.length > 0) {
+        (element as HTMLElement).style.animationPlayState = 'running'; }
+      });
+    }
+
+    return generatePublicAPI();
+  };
+
+  const stop = (): Animation => {
+    childAnimations.forEach(animation => {
+      animation.stop();
+    });
+
+    if (initialized) {
+      cleanUp();
+      initialized = false;
+    }
+
+    return generatePublicAPI();
+  };
+
+  const from = (property: string, value: any): Animation => {
+    const keyframeValues = getKeyframes();
+    const firstFrame = keyframeValues[0];
+
+    if (firstFrame != null && (firstFrame.offset === undefined || firstFrame.offset === 0)) {
+      firstFrame[property] = value;
+    } else {
+      const object: any = {
+        offset: 0
+      };
+      object[property] = value;
+
+      _keyframes = [
+        object,
+        ..._keyframes
+      ];
+    }
+
+    return generatePublicAPI();
+  };
+
+  const to = (property: string, value: any): Animation => {
+
+    const keyframeValues = getKeyframes();
+    const lastFrame = keyframeValues[keyframeValues.length - 1];
+
+    if (lastFrame != null && (lastFrame.offset === undefined || lastFrame.offset === 1)) {
+        lastFrame[property] = value;
+    } else {
+
+      const object: any = {
+        offset: 1
+      };
+      object[property] = value;
+
+      _keyframes = [
+        ..._keyframes,
+        object
+      ];
+    }
+
+    return generatePublicAPI();
+  };
+
+  const fromTo = (property: string, fromValue: any, toValue: any): Animation => {
+    return from(property, fromValue).to(property, toValue);
+  };
+
+  const generatePublicAPI = (): Animation => {
```

Comment:
ok! maybe we need some `function` instead of arrow functions, this way we can return `this` instead of calling `generatePublicAPI` everything, ideally, we shold be able to reuse the same object

```
const a = b.from();
a === b
```